### PR TITLE
Support of sbt-web 1.5.x and Play 2.9.x+

### LINKED ISSUE #5: Please update for sbt-web 1.5.x
This plugin does not work with Play Framework 2.9.0 anymore
Please update
Here is stacktrace:
```
[info] Typescript compiling on 6 source(s)
[error] stack trace is suppressed; run 'last Web-assets / typescript' for the full output
[error] (Web-assets / typescript) java.lang.NoSuchMethodError: 'java.lang.Object com.typesafe.sbt.web.SbtWeb$.withActorRefFactory(sbt.State, java.lang.String, scala.Function1)'
ERROR a.a.ActorSystemImpl - Internal server error, sending 500 response
scala.MatchError: java.lang.NoSuchMethodError: 'java.lang.Object com.typesafe.sbt.web.SbtWeb$.withActorRefFactory(sbt.State, java.lang.String, scala.Function1)' (of class java.lang.NoSuchMethodError)
        at play.sbt.run.PlayReload$.$anonfun$taskFailureHandler$8(PlayReload.scala:92)
        at scala.Option.map(Option.scala:230)
        at play.sbt.run.PlayReload$.taskFailureHandler(PlayReload.scala:92)
        at play.sbt.run.PlayReload$.$anonfun$compile$3(PlayReload.scala:144)
        at scala.util.Either$LeftProjection.map(Either.scala:573)
        at play.sbt.run.PlayReload$.compile(PlayReload.scala:144)
        at sbt.PlayRun$.$anonfun$playRunTask$5(PlayRun.scala:99)
        at play.runsupport.Reloader.$anonfun$reload$1(Reloader.scala:493)
        at play.runsupport.Reloader$$anon$1.run(Reloader.scala:58)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
```

### PR DESCRIPTION
**Pull request by PromanSEW**
Fixes #5

### CODE REVIEW

**Comment by PromanSEW** [comment]
Breaking change that fixes this PR:
https://github.com/sbt/sbt-js-engine/pull/66

**Comment by platypii** [comment]
I checked out this PR, did `sbt publishLocal` and then tried upgrading one of my Play projects to 2.9.0. Unfortunately it then hangs when generating my typescript assets with `sbt assets`. I'm investigating why, but it's not giving me very useful logging information. I appreciate the PR, I will continue digging.

**Comment by PromanSEW** [comment]
In my project it compiles successfully, so I don't know where is the problem :thinking:

**Comment by PromanSEW** [comment]
I use sbt run and open `localhost:9000`
Then it compiles automatically

**Comment by platypii** [comment]
It works for the `example/` project in this repo, so I agree that it seems like these changes _basically_ work. But my other Play project is more complicated. I would like to make sure that works before I publish a new version of this library. Actively debugging now.

**Comment by PromanSEW** [comment]
I checked my active production project, not `example`, but good luck in debugging anyway 👍 🤞 
It would be very good if this PR will be merged.
But, I don't know how to keep backward compat, maybe for pre-2.9.0 just use old release? (4.6.4)
If so, should be noted in README

**Comment by platypii** [comment]
Update: it appears to be working but EXTREMELY slowly. I have a project with 151 typescript files, and it's taken over 17 minutes to generate the assets, whereas before this update it took **8 seconds**. Continuing to investigate

Screenshot 2023-11-21 at 09 50 30

**Review by platypii — CHANGES_REQUESTED** [review]
The slowdown is happening because of a change in behavior with this PR.

Previously, `node` was called exactly once, with a list of all the `.ts` files to compile passed as arguments. With this PR, it is calling `node` once for every `.ts` file, separately. This leads to extremely poor performance on projects with large number of typescript files.

I assume this has something to do with the switch back to `sbt-js-engine`, but I am not sure exactly what changed.


Update: maybe this is because of the change to `parallelism`?

**Comment by platypii** [comment]
I opened a discussion [here](https://github.com/sbt/sbt-js-engine/pull/66#issuecomment-1821713719) about the removal of `parallelism` from sbt-js-engine. Unless there is a set of options which allows for calling node ONCE with all source files, I'm going to say that we need to revert back to the fork of SbtJsTask in this repo.

File: build.sbt [hunk 1]
```diff
@@ -33,11 +30,12 @@ resolvers ++= Seq(
...
   Resolver.typesafeRepo("releases"),
   Resolver.sbtPluginRepo("releases"),
-  Resolver.sonatypeRepo("releases"),
   Resolver.mavenLocal
 )
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-web" % "1.4.4")
+resolvers ++= Resolver.sonatypeOssRepos("snapshots")
+
+addSbtPlugin("com.github.sbt" % "sbt-js-engine" % "1.3.0")
```
Comment (mkurz) [inline build.sbt]:
```suggestion
addSbtPlugin("com.github.sbt" % "sbt-js-engine" % "1.3.5-M1")
```

@platypii Please give this a try (as soon as it is available at https://repo1.maven.org/maven2/com/github/sbt/sbt-js-engine_2.12_1.0/) and let me know if it fixes your problem, see https://github.com/sbt/sbt-js-engine/pull/66#issuecomment-1823581550
Thanks!

Comment (platypii) [reply build.sbt]: Confirmed that 1.3.5-M1 does fix the performance regression, and now things are super fast! THANK YOU!! 👏

File: build.sbt [hunk 2]
```diff
@@ -33,11 +30,12 @@ resolvers ++= Seq(
...
   Resolver.typesafeRepo("releases"),
   Resolver.sbtPluginRepo("releases"),
-  Resolver.sonatypeRepo("releases"),
   Resolver.mavenLocal
 )
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-web" % "1.4.4")
+resolvers ++= Resolver.sonatypeOssRepos("snapshots")
+
+addSbtPlugin("com.github.sbt" % "sbt-js-engine" % "1.3.5-M1")
```
Comment (mkurz) [inline build.sbt]: Please [upgrade](https://repo1.maven.org/maven2/com/github/sbt/sbt-js-engine_2.12_1.0/1.3.5/):
```suggestion
addSbtPlugin("com.github.sbt" % "sbt-js-engine" % "1.3.5")
```

File: build.sbt (hunk 1 above)
Comment (mkurz) [reply build.sbt]: Great! 1.3.5 available now :wink:

File: build.sbt [hunk 3]
```diff
@@ -33,11 +30,12 @@ resolvers ++= Seq(
   Resolver.bintrayRepo("webjars", "maven"),
   Resolver.typesafeRepo("releases"),
   Resolver.sbtPluginRepo("releases"),
-  Resolver.sonatypeRepo("releases"),
   Resolver.mavenLocal
 )
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-web" % "1.4.4")
+resolvers ++= Resolver.sonatypeOssRepos("snapshots")
```
Comment (mkurz) [inline build.sbt]: Why is that snapshots repo needed?

File: build.sbt [hunk 4]
```diff
@@ -33,11 +30,12 @@ resolvers ++= Seq(
   Resolver.bintrayRepo("webjars", "maven"),
   Resolver.typesafeRepo("releases"),
   Resolver.sbtPluginRepo("releases"),
-  Resolver.sonatypeRepo("releases"),
   Resolver.mavenLocal
 )
```
Comment (mkurz) [inline build.sbt]: Pretty sure you can remove *all* resolvers here? Specially
```sbt
Resolver.bintrayRepo("webjars", "maven"),
Resolver.typesafeRepo("releases"),
Resolver.sbtPluginRepo("releases"),
```
are not needed anymore

**Comment by mkurz** [comment]
@PromanSEW You should also upgrade sbt in **all** scripted tests:
https://github.com/platypii/sbt-typescript/blob/master/src/sbt-test/sbt-typescript/testassets/project/build.properties

And also remove all resolvers for **all** scripted tests as well:
https://github.com/platypii/sbt-typescript/blob/master/src/sbt-test/sbt-typescript/testassets/project/plugins.sbt

File: build.sbt (hunk 3 above)
Comment (PromanSEW) [reply build.sbt]: Just fix deprecation, I don't know, this repo was here before

Comment (mkurz) [reply build.sbt]: You can remove it, I am pretty sure

File: build.sbt [hunk 5]
```diff
@@ -19,25 +19,16 @@ scalacOptions ++= Seq(
...
 
-resolvers ++= Seq(
-  Resolver.bintrayRepo("webjars", "maven"),
-  Resolver.typesafeRepo("releases"),
-  Resolver.sbtPluginRepo("releases"),
-  Resolver.sonatypeRepo("releases"),
-  Resolver.mavenLocal
-)
+resolvers += Resolver.mavenLocal
```
Comment (mkurz) [inline build.sbt]: I am pretty sure you can also remove this mavenLocal repo.

**Review by platypii — APPROVED** [review]
Looks great, thanks for the changes @PromanSEW and @mkurz !

**Pull request merged**